service/src/main/kotlin/fi/espoo/evaka/messaging/MessageQueries.kt

```diff
@@ -325,14 +373,64 @@ FROM message_thread_participant tp
 JOIN message_thread t on t.id = tp.thread_id
 WHERE
     tp.participant_id = :accountId AND
-    tp.last_received_timestamp IS NOT NULL
+    tp.last_received_timestamp IS NOT NULL AND
+    tp.folder_id IS NULL
+ORDER BY tp.last_received_timestamp DESC
+LIMIT :pageSize OFFSET :offset
+        """
+            )
+            .bind("accountId", accountId)
+            .bind("pageSize", pageSize)
+            .bind("offset", (page - 1) * pageSize)
+            .mapToPaged<ReceivedThread>(pageSize)
+
+    val messagesByThread = getThreadMessages(accountId, threads.data.map { it.id })
+    return combineThreadsAndMessages(accountId, threads, messagesByThread)
+}
+
+/** Return all threads in a specific folder in the account */
+fun Database.Read.getThreadsInFolder(
+    accountId: MessageAccountId,
+    pageSize: Int,
+    page: Int,
+    folderName: String
```

Comment:
If you searched by folder ID instead of folder name, you wouldn't need to join on the folder, so querying for `NULL` would be easier. I think it would be better if folder id would be passed all the way from the frontend, and the folder name would be used for viewing purposes only.